pkg/agent/datamodel/types.go

```diff
@@ -1198,7 +1209,9 @@ func (a *AgentPoolProfile) GetKubernetesLabels() string {
 	}
 	sort.Strings(keys)
 	for _, key := range keys {
-		buf.WriteString(fmt.Sprintf(",%s=%s", key, a.CustomNodeLabels[key]))
+		labelName := truncateString(key, MaxK8sLabelNameLength)
+		labelValue := truncateString(a.CustomNodeLabels[key], MaxK8sLabelValueLength)
+		buf.WriteString(fmt.Sprintf(",%s=%s", labelName, labelValue))
```

Comment:
we also add labels within the provisioning scripts themselves, we should take that into account here as well

Reply:
oh nevermind I misunderstood, this on a per-label basis, not total length of the entire set of labels